### PR TITLE
Fix issue where Rust cross compile target is ignored

### PR DESCRIPTION
**Pull request by eberkund**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

I have tested that I can still compile the snap for my native arch and as well as fixing the issue I was having. Please see https://forum.snapcraft.io/t/bug-rust-plugin-not-using-correct-pkg-config-path/7302/4 for more information.

### CODE REVIEW

**Comment by eberkund** [comment]
Just FYI, I believe the tests that are failing are because they are timing out and not because of anything to do with the PR contents.

**Comment by eberkund** [comment]
Thanks for reviewing this.

The problem was that since I had made it so that snapcraft does not write out a `.cargo/config` file, if one is not provided then it fails when cross compiling. I have modified my PR so that it will only write out the config file (same as original behavior) if a config file does not already exist. If a `.cargo/config` file is already provided by the project, then it will use that.

File: tests/unit/plugins/test_rust.py [hunk 1]
```diff
@@ -104,8 +104,6 @@ def test_cross_compile(self, mock_download):
         )
 
         plugin.build()
-        self.assertThat(os.path.join(plugin._cargo_dir, "config"), FileExists())
```
Comment (sergiusens) [inline tests/unit/plugins/test_rust.py]: Given the recent changes, do you mind adding this check back?

**Comment by eberkund** [comment]
Added back `self.assertThat(os.path.join(plugin._cargo_dir, "config"), FileExists())`

**Pull request merged**